invenio_vcs/models.py

```diff
@@ -0,0 +1,336 @@
+# -*- coding: utf-8 -*-
+# This file is part of Invenio.
+# Copyright (C) 2025 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Models for the VCS integration."""
+
+import uuid
+from enum import Enum
+
+from invenio_accounts.models import User
+from invenio_db import db
+from invenio_i18n import lazy_gettext as _
+from invenio_webhooks.models import Event
+from sqlalchemy import UniqueConstraint
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.ext.mutable import MutableDict
+from sqlalchemy_utils.models import Timestamp
+from sqlalchemy_utils.types import ChoiceType, JSONType, UUIDType
+
+RELEASE_STATUS_TITLES = {
+    "RECEIVED": _("Received"),
+    "PROCESSING": _("Processing"),
+    "PUBLISHED": _("Published"),
+    "FAILED": _("Failed"),
+    "DELETED": _("Deleted"),
+}
+
+RELEASE_STATUS_ICON = {
+    "RECEIVED": "spinner loading icon",
+    "PROCESSING": "spinner loading icon",
+    "PUBLISHED": "check icon",
+    "FAILED": "times icon",
+    "DELETED": "times icon",
+}
```

Comment:
I am not 100% sure about this given that it highly depends on the CSS frameworks in use.
Can we map this in the front end instead, via an overridable Jinja macro for example?